src/bindings/python/flux/job.py

```diff
@@ -411,63 +394,53 @@ def event_watch_async(flux_handle, jobid, name="*", eventlog="eventlog"):
     :returns: a JobEventWatchFuture object
     :rtype: JobEventWatchFuture
     """
-    from fnmatch import fnmatch
 
-    flags = 0
-    wait_future = JobEventWatchFuture(flux_handle)
+    future = RAW.event_watch(flux_handle, int(jobid), eventlog, flags)
+    return JobEventWatchFuture(future)
 
-    def event_watch_cb(future, arg):
-        nonlocal wait_future
-        nonlocal name
-        result = ffi.new("char *[1]")
-        try:
-            RAW.event_watch_get(future, result)
-        except OSError as exc:
-            if exc.errno == errno.ENODATA:
-                wait_future.fulfill(None)
-            else:
-                wait_future.fulfill_error(exc.errno, future.error_string())
-            return
-        event = EventLogEvent(ffi.string(result[0]).decode("utf-8"))
-        if fnmatch(event.name, name):
-            wait_future.fulfill(event)
-        future.reset()
 
-    Future(RAW.event_watch(flux_handle, int(jobid), eventlog, flags)).then(
-        event_watch_cb
-    )
+def event_watch(flux_handle, jobid, eventlog="eventlog", flags=0):
```

Comment:
Whatever we decided w.r.t. `flags` for in `event_watch_async` should apply here too.